pandas/tests/io/excel/test_writers.py

```diff
@@ -652,25 +652,27 @@ def test_excel_roundtrip_datetime(self, merge_cells, tsframe, path):
 
         tm.assert_frame_equal(tsframe, recons)
 
-    def test_excel_date_datetime_format(self, ext, path):
+    def test_excel_roundtrip_date_datetime_format(self, ext, path):
```

Comment:
Why are you changing this test?

Reply:
In short, because it fails on its own terms, i.e. it does not test what it purports to. The problem is that each column contains one date and one datetime, so on creation of `df` both columns are set to `datetime64[ns]`. So when it comes to writing to Excel `df` only contains datetimes and therefore the `date_format` parameter doesn't get exercised.

To fix this, I transposed the DataFrame so that there is one column of dates and one column of datetimes, meaning that at write-time there are still dates inthe DataFrame so the date_format parameter is actually being used.

If you're asking why I'm changing this test when it isn't strictly part of my bugfix, it is the only other test for the `date_format` and `datetime_format` parameters, so I looked into it when I was doing the fix to work out why it hadn't caught the bug that had been introduced. In doing so, I noticed that it wasn't quite working as it seemed the author intended (as described above). Since it's also testing these parameters, and it was a small fix, I thought it was reasonable to include it in the same PR.

I changed the name because I initially put my new test in the same test class and was trying to find a different name for it, but felt that it was better having the `test_excel_date_datetime_format` name as it actually verifies the format, whereas "roundtrip" seemed to better describe what this test actually covers.

Reply:
And should I just amend this PR with a commit reverting the change to the test, or is it better to close this and open a new one without that change? This is my first open-source PR so I'm not sure what best practice with this is. Thanks again for your help reviewing 😃
(I say "amend this PR with a commit" - I don't yet know how to do that, but I think I read somewhere that it's possible!)

Reply:
I've reverted this commit with these changes.